Node organization

devspacenine opened the pull request:
Organized the routes so that its easier to add new ones and added a directory for middleware. Also, did a few cleanup things.

chrislbs commented on app.js:
```diff
@@ -1,19 +1,35 @@
+app.set('rootPath', __dirname);
+// Global Paths
+app.set('paths', {
+    routes: path.join(app.get('rootPath'), 'routes'),
```
Can you explain a little bit of what is going on here?

devspacenine replied:
The use of just one const keyword is an optimization trick for v8.

The app.set() functions are creating global values that can be accessed anywhere you have a reference to the app obect.

chrislbs replied:
Is this like a best practice or something? It seems weird to me to have to pass around the app everywhere. I would think it would be simpler to have each module completely encapsulate all of its requirements.

devspacenine replied:
No, I just pass it to routes and middlewares so that each module has the option to attach to either the global app or the apiRouter. That way can can do things like write middleware that only applies to certain routers (e.g. auth middleware for api endpoints).

chrislbs replied:
I think the way I'm doing that in the existing api_routes would effectively accomplish that same feat. If I wanted something to apply to all api routes, I could just specify app.use('/api', middleware) . Then all calls to the api routes would pass through that middleware right?

chrislbs commented on middleware/_throttle.js:
```diff
@@ -0,0 +1,85 @@
+'use strict';
```
This middleware seems like overkill for this app

devspacenine replied:
Oh, it definitely is. I just needed a file in the directory so that it could be included with git

chrislbs replied:
Alright. Doesn't really hurt anything.

chrislbs commented on routes/api/players.js:
```diff
@@ -1,59 +1,57 @@
- * Create a new player
- */
-router.post('/', function(request, response) {
+module.exports = function(app, apiRouter) {
```
Why are we having to pass the node app to these exports now?

devspacenine replied:
So that all of the modules we import have access to "global" app values. Like the paths that are defined in app.js.

chrislbs commented on routes/api/players.js:
```diff
@@ -1,59 +1,57 @@
+    /**
+     * Create a new player
+     */
+    apiRouter.post('/player/', function(request, response) {
```
I generally prefer to use plural names for rest resources as a GET to /players represents a request to the entire collection of players.

devspacenine replied:
righto

devspacenine commented on app.js:
```diff
@@ -1,19 +1,35 @@
+app.use('/api', apiRouter);
 
 app.get('/users', function(request, response) {
 
```
I didn't put these guys in their own module yet, because I wasn't sure what your plan is with them.

chrislbs replied:
Those were just some garbage I threw together as an example. I'd actually go ahead and remove them at some point.

devspacenine commented:
Went ahead and removed the user endpoints. And I changed the static folder to something other than root. That way all of the node code and config files aren't served.
